components/roode/__init__.py

```diff
@@ -16,208 +22,217 @@
 roode_ns = cg.esphome_ns.namespace("roode")
 Roode = roode_ns.class_("Roode", cg.PollingComponent)
 
-CONF_ROI_HEIGHT = "roi_height"
-CONF_ROI_WIDTH = "roi_width"
-CONF_ADVISED_SENSOR_ORIENTATION = "advised_sensor_orientation"
+CONF_AUTO = "auto"
+CONF_ORIENTATION = "orientation"
 CONF_CALIBRATION = "calibration"
-CONF_ROI_CALIBRATION = "roi_calibration"
-CONF_INVERT_DIRECTION = "invert_direction"
-CONF_MAX_THRESHOLD_PERCENTAGE = "max_threshold_percentage"
-CONF_MIN_THRESHOLD_PERCENTAGE = "min_threshold_percentage"
-CONF_MANUAL_THRESHOLD = "manual_threshold"
-CONF_THRESHOLD_PERCENTAGE = "threshold_percentage"
+CONF_DETECTION_THRESHOLDS = "detection_thresholds"
+CONF_DISTANCE_MODE = "distance_mode"
 CONF_I2C_ADDRESS = "i2c_address"
-CONF_SENSOR_MODE = "sensor_mode"
-CONF_MANUAL = "manual"
-CONF_MANUAL_ACTIVE = "manual_active"
-CONF_CALIBRATION_ACTIVE = "calibration_active"
-CONF_TIMING_BUDGET = "timing_budget"
-CONF_SAMPLING = "sampling"
-CONF_SAMPLING_SIZE = "size"
-CONF_ROI = "roi"
-CONF_ROI_ACTIVE = "roi_active"
-CONF_ZONES = "zones"
 CONF_ENTRY_ZONE = "entry"
 CONF_EXIT_ZONE = "exit"
-CONF_SENSOR_OFFSET_CALIBRATION = "sensor_offset_calibration"
-CONF_SENSOR_XTALK_CALIBRATION = "sensor_xtalk_calibration"
-TYPES = [CONF_ADVISED_SENSOR_ORIENTATION, CONF_I2C_ADDRESS]
+CONF_CENTER = "center"
+CONF_MAX = "max"
+CONF_MIN = "min"
+CONF_ROI = "roi"
+CONF_SAMPLING = "sampling"
+CONF_TIMING_BUDGET = "timing_budget"
+CONF_XSHUT = "xshut"
+CONF_XTALK = "xtalk"
+CONF_ZONES = "zones"
+
+Orientation = roode_ns.enum("Orientation")
+ORIENTATION_VALUES = {
+    "parallel": Orientation.Parallel,
+    "perpendicular": Orientation.Perpendicular,
+}
+
+DistanceMode = cg.global_ns.enum("EDistanceMode")
+DISTANCE_MODES = {
+    CONF_AUTO: CONF_AUTO,
+    "short": DistanceMode.Short,
+    "long": DistanceMode.Long,
+}
+
+TIMING_BUDGET_ALIASES = {
+    CONF_AUTO: CONF_AUTO,
+    "shortest": 15,
+    "short": 20,
+    "medium": 33,
+    "medium long": 50,
+    "long": 100,
+    "longer": 200,
+    "longest": 500,
+    "15": 15,
+    "20": 20,
+    "33": 33,
+    "50": 50,
+    "100": 100,
+    "200": 200,
+    "500": 500,
```

Comment:
```suggestion
```

I do not want to expose `500` because it does work with people counting. Its just way to slow.